src/main/java/io/temporal/internal/sync/DeterministicRunnerImpl.java

```diff
@@ -71,6 +71,10 @@
 /** Throws Error in case of any unexpected condition. It is to fail a decision, not a workflow. */
 class DeterministicRunnerImpl implements DeterministicRunner {
 
+  private static final int ROOT_THREAD_PRIORITY = 0;
+  private static final int CALLBACK_THREAD_PRIORITY = 10;
+  private static final int WORKFLOW_THREAD_PRIORITY = 20000000;
```

Comment:
These priorities are not Java thread priorities. They define the order of cooperative scheduling to ensure that signals are processed before the main method. And for the root thread we need it to ensure that it runs before the first signal is delivered in case of signalWithStart.